lib/modules/manager/poetry/schema.ts

```diff
@@ -132,58 +132,152 @@ export const PoetryDependencyRecord = LooseRecord(
     dep.skipReason = 'invalid-version';
     return dep;
   })
-).transform((record) => {
-  for (const [depName, dep] of Object.entries(record)) {
-    dep.depName = depName;
-    if (!dep.packageName) {
-      const pep503NormalizeRegex = regEx(/[-_.]+/g);
-      const packageName = depName
-        .toLowerCase()
-        .replace(pep503NormalizeRegex, '-');
-      if (depName !== packageName) {
-        dep.packageName = packageName;
+)
+  .transform((record) => {
+    const deps: PackageDependency[] = [];
+    for (const [depName, dep] of Object.entries(record)) {
+      dep.depName = depName;
+      if (!dep.packageName) {
+        const pep503NormalizeRegex = regEx(/[-_.]+/g);
+        const packageName = depName
+          .toLowerCase()
+          .replace(pep503NormalizeRegex, '-');
+        if (depName !== packageName) {
+          dep.packageName = packageName;
+        }
       }
+      deps.push(dep);
     }
-  }
-  return record;
-});
+    return deps;
+  })
+  .catch([]);
```

Comment:
still the question why catch here? do we really need to be so defensive? I would suggest to log a debug log with failed file parsing

Reply:
For example, if we have `dependencies` field, but `devDependencies` is absent, we don't want to fail the entire parsing process. This schema is being used for 3 fields that aren't guaranteed to be present. I don't think we need to log their absense.

Reply:
Okay, I've changed it to something more clear